### PR TITLE
Fix Asciidoctor properties warnings

### PR DESCRIPTION
**Pull request by lptr**

### CODE REVIEW

**Comment by lptr** [comment]
Feedback build: https://builds.gradle.org/viewQueued.html?itemId=10246016

File: buildSrc/src/main/groovy/org/gradle/build/docs/CacheableAsciidoctorTask.groovy [hunk 1]
```diff
@@ -15,33 +15,129 @@
...
+    @PathSensitive(PathSensitivity.RELATIVE)
+    @Override
+    File getBaseDir() {
+        return super.getBaseDir()
+    }
+
+    @Classpath
+    @Optional
+    @Override
+    Configuration getClasspath() {
```
Comment (wolfs) [inline buildSrc/src/main/groovy/org/gradle/build/docs/CacheableAsciidoctorTask.groovy]: The task action does the following:
```
if (classpath == null) {
    classpath = project.configurations.getByName(AsciidoctorPlugin.ASCIIDOCTOR)
}
```

That is not very nice :(

Should this method return that instead:
```
super.getClasspath() == null ? project.configurations.getByName(AsciidoctorPlugin.ASCIIDOCTOR) : super.getClasspath()
```

Comment (lptr) [reply buildSrc/src/main/groovy/org/gradle/build/docs/CacheableAsciidoctorTask.groovy]: There are so many things to fix about this task, I don't think it makes sense to fix this single problem. The goal of `CacheableAsciidoctorTask` is to make it work for us in any case. We should instead send a PR to Asciidoctor with fixes.

Comment (wolfs) [reply buildSrc/src/main/groovy/org/gradle/build/docs/CacheableAsciidoctorTask.groovy]: 👍

**Pull request merged**